pkg/agent/monitortool/monitor_test.go

```diff
@@ -160,7 +162,8 @@ func newTestMonitor(
 	crdClientset := fakeversioned.NewSimpleClientset(crdObjects...)
 	crdInformerFactory := crdinformers.NewSharedInformerFactory(crdClientset, 0)
 	nlmInformer := crdInformerFactory.Crd().V1alpha1().NodeLatencyMonitors()
-	m := NewNodeLatencyMonitor(nodeInformer, nlmInformer, nodeConfig, trafficEncapMode)
+	antreaClientProvider, _ := client.NewAntreaClientProvider(componentbaseconfig.ClientConnectionConfiguration{}, clientset)
```

Comment:
That won't work. You need a fake provider that will return the fake client when called. You can check out the test code in pkg/agent/controller/egress/egress_controller_test.go

Reply:
@antoninbas Ok, I configured gomock to generate the fake client and return the clientset using EXCEPT.

But in `monitor.go`->`report()` I need `pkg/client/clientset/versioned/fake/clientset_generated.go`, I need to use antrea's fakeclientset method to get the ` StatsV1alpha1()`.

Do I need to replace the clientset in `k8s.io/client-go/kubernetes/fake` used in test code?

```go
clientset := fake.NewSimpleClientset(objects...)
statsClientset := statsfake.NewSimpleClientset(objects...)
```

Reply:
It's a bit overkill to generate a gomock implementation for that interface. You can just use a few lines to achieve it:
```go
type antreaClientGetter struct {
	clientset versioned.Interface
}

func (g *antreaClientGetter) GetAntreaClient() (versioned.Interface, error) {
	return g.clientset, nil
}

antreaClientProvider := &antreaClientGetter{fakeversioned.NewSimpleClientset(objects...)}
m := NewNodeLatencyMonitor(antreaClientProvider, nodeInformer, nlmInformer, nodeConfig, trafficEncapMode)
```

You don't need to change other clientsets.

Reply:
Ok, that's helpful, I've updated it.